src/test/rgw/bucket_notification/test_bn.py

```diff
@@ -0,0 +1,2777 @@
+                remaining_retries -= 1
+                print('failed to connect to kafka (remaining retries '
+                    + str(remaining_retries) + '): ' + str(error))
+                time.sleep(1)
+
+        if remaining_retries == 0:
+            raise Exception('failed to connect to kafka - no retries left')
+
+        self.events = []
+        self.topic = topic
+        self.stop = False
+
+    def verify_s3_events(self, keys, exact_match=False, deletions=False):
+        """verify stored s3 records agains a list of keys"""
+        verify_s3_records_by_elements(self.events, keys, exact_match=exact_match, deletions=deletions)
+        self.events = []
+
+def kafka_receiver_thread_runner(receiver):
+    """main thread function for the kafka receiver"""
+    try:
+        log.info('Kafka receiver started')
+        print('Kafka receiver started')
+        while not receiver.stop:
+            for msg in receiver.consumer:
+                receiver.events.append(json.loads(msg.value))
+            timer.sleep(0.1)
+        log.info('Kafka receiver ended')
+        print('Kafka receiver ended')
+    except Exception as error:
+        log.info('Kafka receiver ended unexpectedly: %s', str(error))
+        print('Kafka receiver ended unexpectedly: ' + str(error))
+
+
+def create_kafka_receiver_thread(topic, security_type='PLAINTEXT'):
+    """create kafka receiver and thread"""
+    receiver = KafkaReceiver(topic, security_type)
+    task = threading.Thread(target=kafka_receiver_thread_runner, args=(receiver,))
+    task.daemon = True
+    return task, receiver
+
+def stop_kafka_receiver(receiver, task):
+    """stop the receiver thread and wait for it to finis"""
+    receiver.stop = True
+    task.join(1)
+    try:
+        receiver.consumer.close()
+    except Exception as error:
+        log.info('failed to gracefuly stop Kafka receiver: %s', str(error))
+
+
+def get_ip():
+    return 'localhost'
+
+def connection():
+    hostname = get_config_host()
+    port_no = get_config_port()
+    vstart_access_key = get_access_key()
+    vstart_secret_key = get_secret_key()
+
+    conn = S3Connection(aws_access_key_id=vstart_access_key,
+	    	      aws_secret_access_key=vstart_secret_key,
+                      is_secure=False, port=port_no, host=hostname, 
+                      calling_format='boto.s3.connection.OrdinaryCallingFormat')
+
+    return conn
+
+def connection2():
+    vstart_access_key = '0555b35654ad1656d804'
+    vstart_secret_key = 'h7GhxuBLTrlhVUyxSPUKUV8r/2EI4ngqJxD7iBdBYLhwluN30JaT3Q=='
+    hostname = get_ip()
+
+    conn = S3Connection(aws_access_key_id=vstart_access_key,
+                      aws_secret_access_key=vstart_secret_key,
+                      is_secure=False, port=8001, host=hostname,
+                      calling_format='boto.s3.connection.OrdinaryCallingFormat')
+
+    return conn
+
+
+##############
+# bucket notifications tests
+##############
+
+'''
+def test_ps_s3_topic_on_master():
```

Comment:
why is this commented?

Reply:
This test contains "get_tenant" call so as we discussed I've skipped this for now.